api/tests/mocha/controllers/forms.spec.js

```diff
@@ -148,4 +152,50 @@ describe('forms controller', () => {
       });
     });
   });
+
+  describe('validate', () => {
+    it('returns ok when validations passed', async () => {
+      const req = { body: '<xml></xml>' };
+      const json = sinon.stub(res, 'json');
+      await controller.__with__({
+        'generateXform': { generate: sinon.stub().resolves() },
+        'auth': { check: sinon.stub().resolves() },
+      })(async () => {
+        await controller.validate(req, res);
```

Comment:
I think it would be safe to add an assertion to test which permissions are required for this endpoint.

Reply:
Done, and I also added in the response the HTTP code returned by the auth module instead of hardcode `400`.

Reply:
I think I might be missing it. What I meant was that we should test with permission with pass to auth.check (in this case `can_configure`).

Reply:
The test is the following: https://github.com/medic/cht-core/pull/6947/files#diff-052cf204502838c62ddc7bf2ea55397365c1bcce1c9c7d6f8d8e62252ca4dc4fR202-R216

Because the component I'm testing is `forms.validate`, not `auth.check`, I mocked the last with the error that is triggered when not the right permission is provided: `'auth': { check: sinon.stub().rejects({ code: 403, message: 'Insufficient privileges' }) }`

Reply:
What I meant is that we should check which auth permissions we require for this endpoint.

Reply:
Adding an assertion with the params we pass to `auth.check` would suffice.

Reply:
Added.